packages/grant-explorer/src/features/round/ViewProjectDetails.tsx

```diff
@@ -83,13 +84,18 @@ export default function ViewProjectDetails() {
   const currentTime = new Date();
   const isBeforeRoundEndDate = round && round.roundEndTime > currentTime;
   const isAfterRoundEndDate = round && round.roundEndTime <= currentTime;
-  const [cart, handleAddProjectsToCart, handleRemoveProjectsFromCart] =
-    useCart();
+  const { projects, add, remove } = useCartStorage();
 
-  const isAlreadyInCart = cart.some(
+  const isAlreadyInCart = projects.some(
     (project) => project.grantApplicationId === applicationId
   );
 
+  const cartProject = projectToRender as CartProject;
```

Comment:
`projectToRender` can be `undefined`, casting will hide that condition.

Reply:
OOS, added todo.